doc/source/serve/http-servehandle.rst

```diff
@@ -160,13 +160,36 @@ HTTP Adapters
 ^^^^^^^^^^^^^
 
 HTTP adapters are functions that convert raw HTTP request to Python types that you know and recognize. 
-You can use it in three different scenarios:
+Its input arguments should be type annotated. At minimal, it should accept a ``starlette.requests.Request`` type.
+But it can also accept any type that's recognized by the FastAPI's dependency injection framework. 
+
+For example, here is an adapter that extra the json content from request. 
+
+.. code-block:: python
+
+    async def json_resolver(request: starlette.requests.Request):
+        return await request.json()
+
+Here is an adapter that accept two HTTP query parameters.
+
+.. code-block:: python
+
+    def parse_query_args(field_a: int, field_b: str):
+        return YourDataClass(field_a, field_b)
```

Comment:
can we put up a full, short snippet here to show user what the http request look like and how its resolved by http_adapter function into `YourDataClass` ?